tests/test_parser/test_properties/test_init.py

```diff
@@ -31,13 +31,35 @@ def test_get_type_string(self, mocker):
         assert p.get_type_string(no_optional=True) == base_type_string
 
         p = Property(name="test", required=False, default=None, nullable=True)
-        assert p.get_type_string() == f"Union[Unset, Optional[{base_type_string}]]"
+        assert p.get_type_string() == f"Union[Unset, None, {base_type_string}]"
         assert p.get_type_string(no_optional=True) == base_type_string
 
         p = Property(name="test", required=False, default=None, nullable=False)
         assert p.get_type_string() == f"Union[Unset, {base_type_string}]"
         assert p.get_type_string(no_optional=True) == base_type_string
 
+    def test_get_type_string_query_parameter(self, mocker):
+        from openapi_python_client.parser.properties import Property
+
+        mocker.patch.object(Property, "_type_string", "TestType")
+        p = Property(name="test", required=True, default=None, nullable=False)
+
+        base_type_string = f"TestType"
+
+        assert p.get_type_string(query_parameter=True) == base_type_string
+
+        p = Property(name="test", required=True, default=None, nullable=True)
+        assert p.get_type_string(query_parameter=True) == f"Optional[{base_type_string}]"
+        assert p.get_type_string(no_optional=True, query_parameter=True) == base_type_string
+
+        p = Property(name="test", required=False, default=None, nullable=True)
+        assert p.get_type_string(query_parameter=True) == f"Union[Unset, None, {base_type_string}]"
+        assert p.get_type_string(no_optional=True, query_parameter=True) == base_type_string
+
+        p = Property(name="test", required=False, default=None, nullable=False)
+        assert p.get_type_string(query_parameter=True) == f"Union[Unset, None, {base_type_string}]"
+        assert p.get_type_string(no_optional=True, query_parameter=True) == base_type_string
```

Comment:
This feels like maybe a good opportunity to use `@pytest.mark.parametrize` to make the different test cases and expected outcomes easier to read. Not at all required (clearly lots of other tests could use it but don't) but could be nice.